app/services/workflow_executions/cancellation_service.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module WorkflowExecutions
+  # Service used to Cancel a WorkflowExecution
+  class CancellationService < BaseService
+    def initialize(workflow_execution, wes_connection, user = nil, params = {})
+      super(user, params)
+      @workflow_execution = workflow_execution
+      @wes_client = Integrations::Ga4ghWesApi::V1::Client.new(conn: wes_connection)
+    end
+
+    def execute
+      return false unless @workflow_execution.submitted?
+
```

Comment:
As per the issue #248 we want to validate that the workflow is in the `cancelling` state before making the call to the WES api to actually cancel the workflow. We will need to update the test as well to assert that it's in the `cancelling` state before calling the cancellation service

Reply:
Fixed in https://github.com/phac-nml/irida-next/pull/328/commits/eb38099002599a4df1ae9329fb92ab779c612d95